functional-programming-spreadsheet/index885.js

```diff
@@ -0,0 +1,117 @@
+const infixToFunction = {
+  "+": (x, y) => x + y,
+  "-": (x, y) => x - y,
+  "*": (x, y) => x * y,
+  "/": (x, y) => x / y
+};
+
+const infixEval = (str, regex) =>
+  str.replace(regex, (_, arg1, fn, arg2) =>
+    infixToFunction[fn](parseFloat(arg1), parseFloat(arg2))
+  );
+
+const highPrecedence = str => {
+  const regex = /([0-9.]+)([*\/])([0-9.]+)/;
+  const str2 = infixEval(str, regex);
+  return str === str2 ? str : highPrecedence(str2);
+};
+
+const isEven = num => num % 2 === 0;
+
+const spreadsheetFunctions = {
+  "": x => x,
+  random: ([x, y]) => Math.floor(Math.random() * y + x),
+  increment: nums => nums.map(x => x + 1),
+  firsttwo: arr => arr.slice(0, 2),
+  lastttwo: arr => arr.slice(-2),
+  even: nums => nums.filter(isEven),
+  sum: nums => nums.reduce((a, x) => a + x),
+  has2: arr => arr.includes(2)
+};
+
+/*
+The `reduce` method can take a second argument (in addition to the function), specifying the initial accumulator value.
+In this case, the current value starts from index 0 rather than index 1.
+```
+[1, [1, 2, 3], [3, 4, 5]].reduce((a, x) => a.concat(x), []) === [1, 1, 2, 3, 3, 4, 5]
+// without the second argument, it first tries 1.concat([1, 2, 3]), but 1 is not an array
+// now it first tries [].concat(1) which works 
+```
+Use this syntax and the `includes` method to add a function `nodups` to `spreadsheetFunctions` which returns the array with duplicates removed.
```

Comment:
This could be split into two steps. The first could instruct the learner to write `nodups: arr => arr.reduce((a, x) => a.includes(x))`, and the next could be about the ternary operator.